docs/troubleshooting-guide/my-dashboard-is-slow.md

```diff
@@ -0,0 +1,125 @@
+
+Note: you may also see your database being overloaded if you're using the same database for Metabase's app database and for your own data. We strongly recommend that you don't do this in a production system or if you have more than a handful of users. If you are doing this, set up a second database for Metabase's own use.
+
+## Are some of your queries intrinsically slow?
+
+**Root cause:** Joining half a dozen tables, each with a few million rows, simply takes a lot of time. While we do our best to create fully-formed SQL queries from graphical questions, SQL snippets, and questions that use other questions as starting points, it's a hard problem---particularly across as many databases as we support.
+
+**Steps to take:**
+
+1. Run each question in the dashboard on its own to see which ones are slow.
+2. Look at the size of the result set for each question that is slow. If it contains hundreds of columns or many thousands of rows, see if you can modify the question to return a smaller result set.
+3. Run the same SQL that Metabase is running but using a different tool. For example, if you have created a question using the Notebook Editor and you are using Postgres as your database, you can view the SQL, copy it, and run it from the command line using `psql`. (If you have written the question in SQL you can just copy and paste it.) If the query runs noticeably faster this way than it does when you run it through Metabase, the problem is almost certainly one of the ones described above.
+4. Run the query with a JDBC-based tool like [DBeaver][dbeaver], which is also Java-based and uses most of the same database drivers as Metabase. Using this, and running the test query from the same machine that Metabase is running on, can help you determine if the problem is a network bottleneck.
+5. If the query runs slowly when sent by another tool, see if you can write a SQL query that calculates the same result as the question you have built in Metabase, but does so more quickly.
+
+## Are we generating inefficient SQL?
+
+**Root cause:** We save questions created graphically in Metabase Query Language (MBQL), then translate MBQL into queries for particular back-end databases. We create the most efficient queries we can, but for the sake of portability, we don't take advantage of every database's idiosyncracies, so sometimes a GUI question will be slower than the equivalent hand-written SQL.
+
+**Steps to take:**
+
+1. Check if you have the most recent version of Metabase: we fix problems as they're reported, and updating Metabase may make your problem go away.
+2. You can use your SQL in place of the code we generate, and [make its result available][organizing-sql] to people who prefer the Notebook Editor as a starting point for their questions.
+3. Please file a bug report to help us figure out how to generate better SQL.
```

Comment:
Link to https://www.metabase.com/docs/latest/troubleshooting-guide/bugs.html?

Reply:
added